### PR TITLE
set up shopify app to build a local optimizely sdk from source

### PR DESCRIPTION
**Pull request by wangjoshuah**

### CODE REVIEW

File: MobileBuy/build.gradle [hunk 1]
```diff
@@ -7,12 +7,17 @@ buildscript {
...
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.2'
+        classpath 'com.android.tools.build:gradle:2.2.3'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.2'
         classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
+
+    // keep this here to allow ENV vairables to be set.
```
Comment (mikeproeng37) [inline MobileBuy/build.gradle]: typo in `variables`

File: MobileBuy/sample/shop.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
 SHOP_DOMAIN=argyle-bliss.myshopify.com
-API_KEY=
-APP_ID=8
\ No newline at end of file
+API_KEY=30e8a770c793382de7516f5fc0a497f1
```
Comment (mikeproeng37) [inline MobileBuy/sample/shop.properties]: this shouldn't be checked in

File: MobileBuy/settings.gradle [hunk 3]
```diff
@@ -1 +1,5 @@
-include ':sample', ':buy'
\ No newline at end of file
+include ':sample', ':buy', ':android-sdk', ':event-handler', ':user-profile', ':shared'
+project(':android-sdk').projectDir = new File('../../android-sdk/android-sdk')
```
Comment (mikeproeng37) [inline MobileBuy/settings.gradle:2]: will the android-sdk included in this repo?

**Review by mikeproeng37 — COMMENTED** [review]
Is the goal of this to make it so that the test app will always build locally as opposed to build from bintray? Is there a reason to skip bintray altogether?

File: MobileBuy/build.gradle (hunk 1 above)
Comment (jophde) [inline MobileBuy/build.gradle]: This really allows the ENV variables to *not* be set.

File: MobileBuy/build.gradle [hunk 4]
```diff
@@ -21,6 +26,25 @@ allprojects {
     }
 }
 
+// snapshot of dependencies to use
```
Comment (jophde) [inline MobileBuy/build.gradle]: This allows the test app to control the versions of the deps that the android sdk is built with without needing to modify code in the android sd repo itself.

**Review by jophde — APPROVED** [review]

File: MobileBuy/sample/shop.properties (hunk 2 above)
Comment (jophde) [reply MobileBuy/sample/shop.properties]: Agree with mike, this should be read in from env variable or be git ignored.

**Review by jophde — CHANGES_REQUESTED** [review]
Shouldn't store API key in source

**Comment by jophde** [comment]
@mikeng13 building from source locally like this allows the android-sdk be checked out at whatever commit.  It could be checked out at a release tag or at a feature branch.  It could be done with bintray and artifacts but you would need push every feature branch or commit you want to test up as artifact to bintray/maven.

**Comment by mikeproeng37** [comment]
@jophde does this mean that we need to update the android-sdk source here in this repo every time we want to test a new version of the SDK? I don't see where we are adding the android-sdk source code to this repo

**Comment by jophde** [comment]
We aren't adding to the repo.  You just to clone this repo and `android-sdk` in the same directory.

**Comment by mikeproeng37** [comment]
Ah got it. #noobquestions

**Review by jophde — APPROVED** [review]

**Pull request merged**